src/CodeGeneration.Roslyn.Tool/Program.cs

```diff
@@ -14,6 +16,24 @@ namespace CodeGeneration.Roslyn.Generate
     internal static class Program
     {
         private static async Task<int> Main(string[] args)
+        {
+            using (var cancellationTokenSource = new CancellationTokenSource())
```

Comment:
I'd suggest to initialize this `cts` besides `progress` and dispose in the `finally` of the existing `try`.

@AArnott what's your opinion?

Reply:
My intention is to separate the cancellation aspect from the rest (here `Core` method)

Reply:
Either way is fine. I'd vote to go with what @manne has already done. This is *very* similar to [my prescriptive pattern](https://gist.github.com/AArnott/daa2e450fb5af165f5ca2aff42a4b5cc).